src/WordConverter.php

```diff
@@ -55,20 +55,21 @@ private function convertUpToLac($amount)
     protected function convertToWord($amount)
     {        
         $word = '';
-        $splitAmounts = preg_split("/\./", round($amount,2));
+        $splitAmounts = preg_split("/\./", number_format($amount,2,'.',''));
         $amount = $splitAmounts[0];        
-   
+        
         if( strlen($amount) >= self::CRORE  ) { // crore
             $chunkLength = (strlen($amount) - ( self::CRORE - 1));
             $chunkAmount = substr($amount, 0, $chunkLength);
             $amount = substr($amount, $chunkLength) ;
             $word .= $this->convertUpToLac($chunkAmount)  . ' crore ';
         }
 
-        $word .= $this->convertUpToLac($amount);
+        $word .= $this->convertUpToLac($amount). ' Taka';
 
         if( isset($splitAmounts[1]) ) {
-            $word .= ' and ' . $this->taka[ $splitAmounts[1]] . ' poisa';
+            if((int)$splitAmounts[1]!=0)
```

Comment:
Working great but curious why you did not use `(int)$splitAmounts[1] > 0` instead of confusing not(!) condition?